### ISSUE TITLE
Error: Cannot convert float NaN to integer

### ISSUE THREAD

**Issue by gszecsenyi** [sdv-dev/SDV#1639]
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 1.5.0
* Python version: 3.9.6
* Operating System: OSX 13.6

### Error Description

On Multi table sampling I get the following error: "Cannot convert float NaN to integer"

### Steps to reproduce

Dataset is from Kaggle: https://www.kaggle.com/datasets/algorismus/adventure-works-in-excel-tables

Code:

reseller = pd.read_csv('aw_dataset_smaller/Reseller.csv', delimiter='\t')
sales = pd.read_csv('aw_dataset_smaller/Sales.csv', delimiter='\t')
sales['OrderDate'] = pd.to_datetime(sales['OrderDate'], format="%A, %B %d, %Y")
product = pd.read_csv('aw_dataset/Product.csv', delimiter='\t')

multi_table_data = {
    'reseller': reseller,
    'sales': sales,
    'product': product,
}

metadata = MultiTableMetadata()

from sdv.multi_table import HMASynthesizer

synthesizer2 = HMASynthesizer(metadata)
synthesizer2.fit(multi_table_data)

result = synthesizer2.sample(scale = 10.0)

**Comment by npatki** [comment]
Hi @gszecsenyi, thanks for filing the issue with the detailed steps. We were able to reproduce this on a similar dataset. 

I'm attaching the trace below for future reference.
[stack_trace.txt](https://github.com/sdv-dev/SDV/files/13034127/stack_trace.txt)

### Root Cause
Based on the error message, it appears that the issue is happening during post-processing, in RDT's UniformEncoder. I believe the root cause is [RDT #719](https://github.com/sdv-dev/RDT/issues/719).

The **good news** is that this was fixed just yesterday. It should be available in the upcoming release.

### Workaround
Until we fix the issue, a workaround is to override the default method for pre- and post-processing. The SDV assigns [UniformEncoder](https://docs.sdv.dev/rdt/transformers-glossary/categorical/uniformencoder) by default. I'd recommend updating it to [LabelEncoder](https://docs.sdv.dev/rdt/transformers-glossary/categorical/labelencoder) (with noise).

Unfortunately, overriding this is a bit manual. You'd have to go through table by table, identify the columns assigned to UniformEncoder, and then manually update them to LabelEncoder.

```python
# start with the 'reseller' table
>>> synthesizer.get_transformers('reseller')
{'ResellerKey': IDGenerator(),
 'Business Type': UniformEncoder(),
 'Reseller': AnonymizedFaker(function_name='bothify', function_kwargs=...),
 'City': AnonymizedFaker(function_name='bothify', function_kwargs=...),
 'State-Province': UniformEncoder(),
 'Country-Region': UniformEncoder()}

# for anything that is assigned UniformEncoder, replace it with LabelEncoder
>>> from rdt.transformers.categorical import LabelEncoder
>>> synthesizer.update_transformers(
        table_name='reseller',
        column_name_to_transformer={
            'Business Type': LabelEncoder(add_noise=True),
            'State-Province': LabelEncoder(add_noise=True),
            'Country-Region': LabelEncoder(add_noise=True)
    })

# TODO repeat this for the remaining tables
```

When you're done, re-fit the synthesizer and sampling should then work.

```python
>>> synthesizer.fit(multi_table_data)
>>> synthetic_data = synthesizer.sample(scale=10.0)
```

**Comment by gszecsenyi** [comment]
Thanks! Great support!

**Comment by golsonallen** [comment]
Hello, 

Was this issue fixed? I'm on version 1.8.0 and am running into this error.

Thanks!

**Comment by npatki** [comment]
Hello, it turns out that although the reverse transformation issue is now resolved in our preprocessing library (RDT), there is a different problem with HMA that is still unresolved. Apologies for the confusion.

It turns out that there is a bug in HMA where it is sometimes creating null values (NaN) even though there are were none in the real data. If the original data is type integer, then HMA tries to convert the NaN back into an integer, which yields this error (int columns are not allowed to be null). This was also described in issue #1730, and the ultimate fix is being tracked in #1691. I'll close this issue as a duplicate.

@golsonallen Until we fix this, please feel free to use the workaround in my above comment or the different workaround in #1730. Both should be able to get you on your way to creating synthetic data.
